aws-sdk-resources/lib/aws-sdk-resources/services/s3/encryption/decrypt_handler.rb

```diff
@@ -31,9 +36,41 @@ def call(context)
 
         def attach_http_event_listeners(context)
           context.http_response.on_headers(200) do
-            cipher = decryption_cipher(context)
+
+            http_resp = context.http_response
             body = context.http_response.body
-            context.http_response.body = IODecrypter.new(cipher, body)
+            cipher = decryption_cipher(context)
+
+            if http_resp.headers['x-amz-meta-x-amz-tag-len']
+
+              content_length = http_resp.headers['content-length'].to_i
+              auth_tag_length = http_resp.headers['x-amz-meta-x-amz-tag-len']
+              auth_tag_length = auth_tag_length.to_i / 8
+
+              auth_tag = context.client.get_object(
```

Comment:
I know currently that we're doing this ranged get call due to the nature of GCM decryption, but a comment explaining why we're doing this (and why we're doing this instead of double-reading the object) would be useful for 1-2 years down the road.